using_restricted_array.rb

```diff
@@ -67,21 +67,30 @@ def find_smallest(array, length)
 
 # Reverses the values in the integer array in place
 def reverse(array, length)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = length - 1
+  while i < (length / 2)
+    temp = array[i]
+    array[i] = array[j]
+    array[j] = temp
+    i += 1
+    j -= 1
+  end
+  return array
 end
 
 # For an array sorted in ascending order, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def binary_search(array, length, value_to_find)
-  asc_array = sort(array, length)
-  low = asc_array[0]
-  high = asc_array[1]
+  # asc_array = sort(array, length)
+  low = 0
```

Comment:
Please take a look at the updated binary_search slide. 
Here's the considerations I was missing in the algorithm on the slide when we discussed it in class, before the update: (1) what if there is only one element in the array and it is the value_to_find? In this case, it used to return false. (To fix this, add an additional if statement after the loop ends, to check if array[low] is the value_to_find.) (2) Consider the scenario where there are two elements in the array and the value_to_find is the second element in the array. In the first iteration, low = 0, high=1, mid=0. Value is not at index 0. So, low = mid (still 0). On re-entering the loop, mid would be still 0. So we had an infinite loop. (To fix this, update high to be mid -1, if value_to_find is less than mid. And update low = mid+1, if value_to_find is greater than mid.)